### PR TITLE
C-API expose QPDFObject::getTypeCode and QPDFObject::getTypeName

### PR DESCRIPTION
**Pull request by m-holger**
Also fix minor typo

### CODE REVIEW

File: include/qpdf/qpdf-c.h [hunk 1]
```diff
@@ -587,6 +587,21 @@ extern "C" {
 
     typedef unsigned int qpdf_oh;
 
+    typedef int QPDF_OT_CODE;
```
Comment (jberkenbilt) [inline include/qpdf/qpdf-c.h:590]: I won't accept this -- it duplicates information. To implement this, I would want to move object_type_e from QPDFObject.hh to Constants.h and just make QPDFObject::object_type_e an alias to it. I can just do that in my current work and then you can redo this PR to use that.

**Review by jberkenbilt — CHANGES_REQUESTED** [review]

File: ChangeLog [hunk 2]
```diff
@@ -1,7 +1,7 @@
 2021-12-16  Jay Berkenbilt  <ejb@ql.org>
 
 	* Add several functions to the C API for working with pages. C
-	wrappers around sevearl of the "Legacy" page operations from
```
Comment (jberkenbilt) [inline ChangeLog:4]: I use a spell checker, which catches this. I will add spell checking to CI.

File: include/qpdf/qpdf-c.h (hunk 1 above)
Comment (jberkenbilt) [reply include/qpdf/qpdf-c.h:590]: My solution seems like it is also going to duplicate information, but I'm going to use it anyway, so please stay tuned.

Comment (m-holger) [reply include/qpdf/qpdf-c.h:590]: I didn't like my solution either, but couldn't think of a better one.

Do we actually need to publish the constants in qpdf-c.h? We could have a simple comment that refers QPDFObject.h

I for one won't use the constants but define my own enum. I suspect most other users will do the same.

Comment (jberkenbilt) [reply include/qpdf/qpdf-c.h:590]: I think it's important to have the enumerated type exposed to the C API if we are going to return values from it. Anyway, I already pushed the change to main.

**Comment by jberkenbilt** [comment]
@m-holger I think this looks good except where we commented. The spelling error is fixed on main already. If you rework this to get rid of the `#define` and new type and use the now exposed `qpdf_object_type_e` as your return type, I will most likely accept this. If you don't get to it, I will take this pull request in its current form and modify it to work so I can get this by the end of the day. I was originally thinking I'd release 10.5 this weekend, but I want to give it a few days for this stuff to sink in. I'll probably aim for early next week. If I don't get it out by around Tuesday, then I will probably want to wait until after the new year.

**Comment by jberkenbilt** [comment]
I've incorporated this.

**Pull request closed without merging**